src/schach.cpp

```diff
@@ -2,7 +2,7 @@
 #include <string>
 #include <vector>
 
-#include "fen.cpp"
```

Comment:
Shouldn't #include cpp files. Put interfaces in .h files and implementations in .cpp files.